src/js/tech/tech.js

```diff
@@ -449,6 +449,42 @@ class Tech extends Component {
            component instanceof Tech ||
            component === Tech;
   }
+
+  /**
+   * Registers a Tech
+   *
+   * @param {String} name Name of the Tech to register
+   * @param {Object} tech The tecj to register
+   * @static
+   * @method registerComponent
+   */
+  static registerTech(name, tech) {
+    if (!Tech.techs_) {
+      Tech.techs_ = {};
+    }
+
+    Tech.techs_[name] = tech;
```

Comment:
Good question.

In the case that we don't necessarily want to inherit from `Tech`, we could do some kind of duck typing to enforce a `tech`'s interface. Of course, that begs the question, do we want to go down that road or just leave it as-is.